Added support for date in cover

SamuelMatamoros opened the pull request:
Added support for date in the cover along the alignment of it in the header.

closes #21

SamuelMatamoros committed:
Added support for including date in the cover

SamuelMatamoros committed:
Simplified syntax

rajayonin commented on lib.typ:
```diff
@@ -163,6 +165,8 @@
 /// - appendixes (content, none): Set of appendixes.
 /// - chapter_on_new_page (bool):  Whether to start each chapter on a new page (`true`) or not (`false`)
+/// - date (content): You can put `#datetime.today().display()` or manually enter the date you wish. By default it is set to none
+/// - date-align (alignment): Alignment of the date inside the header. `right` by default
```
```suggestion
/// - date (content, none): Shows the date in the cover. You can put `#datetime.today().display()` or manually enter the date you wish. If `none` (default), it's not shown.
/// - date-align (alignment): Alignment of the date inside the header. By default, `right`.
```

rajayonin commented:
I don't like that it's shown in the header
<img width="920" height="558" alt="image" src="https://github.com/user-attachments/assets/46de0ee5-999a-47a8-9eba-596b9eb0b69f" />

I was thinking something like above the professor's name...

SamuelMatamoros committed:
Better doc

Co-authored-by: Luis Daniel Casais <71148501+rajayonin@users.noreply.github.com>

rajayonin commented:
And maybe we can do like in the thesis template and ask just for the date object and optionally a formatting, and format the date by default.

rajayonin commented:
We could also allow the user to input content and put that.

rajayonin commented:
> And maybe we can do like in the thesis template and ask just for the date object and optionally a formatting, and format the date by default.

`date` and `date-format`, see [guluc3m/uc3m-thesis-ieee-typst:titlepage.typ#L127-L136](https://github.com/guluc3m/uc3m-thesis-ieee-typst/blob/main/titlepage.typ#L127-L136).

SamuelMatamoros commented:
> I don't like that it's shown in the header
> 
> I was thinking something like above the professor's name...

I agree but i didn't know where to put it...

SamuelMatamoros commented:
> > And maybe we can do like in the thesis template and ask just for the date object and optionally a formatting, and format the date by default.
> 
> `date` and `date-format`, see [guluc3m/uc3m-thesis-ieee-typst:titlepage.typ#L127-L136](https://github.com/guluc3m/uc3m-thesis-ieee-typst/blob/main/titlepage.typ#L127-L136).

I'll check i out

SamuelMatamoros committed:
Changed the date to follow the theesis template style

The date attribute now takes the datetime type and the date format can be specifies with a format string in the date-format attribute

SamuelMatamoros committed:
Added missing comments

SamuelMatamoros committed:
Added support for entering content or a datetime object for the date attribute

SamuelMatamoros commented:
I've made the requested changes, now the user can input either a `datetime` object or just `content` into the `date` attribute. The date-format only applies if the type of `date` is `datetime`.

Still don't know if above the professors name is the best place of the date...

<img width="612" height="474" alt="image" src="https://github.com/user-attachments/assets/9ff1a25e-f678-486c-a47e-0c19e500e245" />

rajayonin commented on lib.typ:
```diff
@@ -46,6 +46,8 @@
 /// - professor (str, none): Professor's name
 /// - team (str): Team name (optional)
 /// - language (str): Report language, either `"es"` or `"en"`
+/// - date (datetime | content): Presentation date. Either a `datetime` object or `content`
```
This isn't Python or TypeScript :)
```suggestion
/// - date (datetime, content): Presentation date. Either a `datetime` object or `content`
```

rajayonin commented on lib.typ:
```diff
@@ -131,8 +140,25 @@
+      date
+    } else {
+      if date-format == auto {
+        date-format = "[Day] [month repr:long] [year]"
```
It should take into account the language.

rajayonin commented on lib.typ:
```diff
@@ -163,6 +189,8 @@
 /// - appendixes (content, none): Set of appendixes.
 /// - chapter_on_new_page (bool):  Whether to start each chapter on a new page (`true`) or not (`false`)
+/// - date (datetime): Presentation date.
+/// - date-format (str): Format syntax (see https://typst.app/docs/reference/foundations/datetime/#format)
```
```suggestion
/// - date (datetime, none): Report date to put in the cover page, e.g. `datetime.today()`. If `none`, no date will be displayed.
/// - date-format (str, auto): Format syntax (see https://typst.app/docs/reference/foundations/datetime/#format). If `auto`, will change with language.
```

SamuelMatamoros committed:
Added support for months in date to be displayed in Spanish

rajayonin commented on lib.typ:
```diff
@@ -189,8 +212,8 @@
-/// - date (datetime): Presentation date.
-/// - date-format (str): Format syntax (see https://typst.app/docs/reference/foundations/datetime/#format)
+/// - date (datetime, none): Report date to put in the cover page, e.g. `datetime.today()`. If `none`, no date will be displayed.
+/// - date-format (str, auto): (Only applies if `date` attribute is a datetime object) Format syntax (see https://typst.app/docs/reference/foundations/datetime/#format). It will change with the language (for supported languages).
```
```suggestion
/// - date-format (str, auto): Date format syntax (see https://typst.app/docs/reference/foundations/datetime/#format). If `auto`, it will be set according to `language`. Only applies if `date` is not `none`.
```

SamuelMatamoros committed:
Changed doc for date-format
